packages/zowe-explorer-api/src/globals/Gui.ts

```diff
@@ -0,0 +1,276 @@
+     *
+     * @see vscode.window.setStatusBarMessage for more details
+     */
+    // Disabling no-explicit-any as it is defined as such within VS Code
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    export function setStatusBarMessage(text: string, timeoutOrThenable?: number | Thenable<any>): vscode.Disposable {
+        if (typeof timeoutOrThenable === "number") {
+            return vscode.window.setStatusBarMessage(text, timeoutOrThenable);
+        }
+        return vscode.window.setStatusBarMessage(text, timeoutOrThenable ?? undefined);
+    }
+
+    /**
+     * Builds the specified QuickPick result based on user interaction.
+     * @param quickpick The QuickPick object to resolve
+     * @returns A promise containing the result of the QuickPick
+     */
+    export function resolveQuickPick(
+        quickpick: vscode.QuickPick<vscode.QuickPickItem>
+    ): Promise<vscode.QuickPickItem | undefined> {
+        return new Promise<vscode.QuickPickItem | undefined>((c) => {
+            quickpick.onDidAccept(() => c(quickpick.activeItems[0]));
+            quickpick.onDidHide(() => c(undefined));
+        });
+    }
+
+    /**
+     *
+     * @param items An array of items, or a promise that resolves to an array of items
+     * @param options VS Code options for the quick pick's behavior
+     * @param token A token used to signal cancellation for the quick pick
+     *
+     * @see vscode.window.showQuickPick for more details
+     */
+    export function showQuickPick<T extends string>(
+        items: readonly T[] | Thenable<readonly T[]>,
+        options?: vscode.QuickPickOptions,
+        token?: vscode.CancellationToken
+    ): Thenable<T | undefined>;
+    export function showQuickPick<T extends string>(
+        items: readonly T[] | Thenable<readonly T[]>,
+        options?: vscode.QuickPickOptions & { canPickMany: true },
+        token?: vscode.CancellationToken
+    ): Thenable<T[] | undefined>;
+    export function showQuickPick<T extends vscode.QuickPickItem>(
+        items: readonly T[] | Thenable<readonly T[]>,
+        options?: vscode.QuickPickOptions,
+        token?: vscode.CancellationToken
+    ): Thenable<T | undefined>;
+    export function showQuickPick<T extends vscode.QuickPickItem>(
+        items: readonly T[] | Thenable<readonly T[]>,
+        options?: vscode.QuickPickOptions & { canPickMany: true },
+        token?: vscode.CancellationToken
+    ): Thenable<T[] | undefined> {
+        return vscode.window.showQuickPick(items, options, token);
+    }
+
+    /**
+     * Displays a message to the user, including any specified options.
+     * @param message The message to display
+     * @param options Any additional options for the message
+     * @returns A thenable containing the selected item (if items were specified), or `undefined`
+     */
+    export function showMessage(message: string, options?: GuiMessageOptions): Thenable<string> {
```

Comment:
Should all exported functions have as few parameters as possible (following this one as an example)?

For instance, from `createWebviewPanel(type, title, showOptions, options)` to `createWebviewPanel(title, options: GuiWebviewPanelOptions)`

I guess I'm hoping that this will be more flexible if we need to add optional properties in the future 😋 

This may apply to other places, and maybe we can refactor some other interfaces and exports to have as few parameters as possible. 😋 

P.S. this is not a request, just a suggestion😋